timed/projects/views.py

```diff
@@ -55,11 +56,27 @@ class BillingTypeViewSet(ReadOnlyModelViewSet):
         # superuser may edit all billing types
         IsSuperUser
         # internal employees may read all billing types
-        | IsAuthenticated & IsInternal & IsReadOnly
+        | IsAuthenticated & (IsInternal | IsCustomer) & IsReadOnly
     ]
 
     def get_queryset(self):
-        return models.BillingType.objects.all()
+        """Get billing types depending on the user's role.
+
+        Internal employees should see all billing types.
+        Customers should only see billing types that are used in customer visible projects.
+        """
+        user = self.request.user
+        queryset = models.BillingType.objects.all()
+
+        current_employment = user.get_active_employment()
+
+        if current_employment is None:
+            if models.CustomerAssignee.objects.filter(
+                user=user, is_customer=True
+            ).exists():
+                return queryset.filter(projects__customer_visible=True)
+        elif not current_employment.is_external:
+            return queryset
```

Comment:
I suggest changing the `if` structure to an early return for the "normal" case. If I see it correctly, the code doesn't properly handle the case where you have an "external" employment - in this case, it returns `None` (should be `queryset.none()`).

Then, the customer assignee should probably be limited to the billing types for the "own" projects only - not that the billing types are actually that critical.

So, we end up with a rough implementation like this:

```python
if current_employment:
    return queryset.none() if current_employment.is_external else queryset

return queryset.filter(
    projects__customer_visible=True,
    projects__customer__assignee__user=user,
    projects__customer__assignee__is_customer=True
)
```

Reply:
Permissions don't allow access for users with external employment. That's why I didn't including handling of that case.
However there is a potential case in which external employment might be relevant: in a case where a customer has an (external) employment.
I'm not sure though whether that's a probable case.